packages/blocks/src/note-block/note-edgeless-block.ts

```diff
@@ -406,6 +409,25 @@ export class EdgelessNoteBlockComponent extends toEdgelessBlockComponent(
 
     const extra = this._editing ? ACTIVE_NOTE_EXTRA_PADDING : 0;
 
+    let backgroundColor = `${DEFAULT_NOTE_BACKGROUND_COLOR}`;
+    if (model.doc.awarenessStore.getFlag('enable_color_picker')) {
+      const rootComponent = this
+        .rootComponent as EdgelessRootBlockComponent | null;
+      if (rootComponent) {
+        backgroundColor =
+          rootComponent.surface.themeObserver.generateColorPropertyWithLightDark(
+            model.background,
+            backgroundColor
+          );
+      }
+    } else if (typeof model.background === 'string') {
+      backgroundColor = model.background;
+    }
+
+    if (backgroundColor.startsWith('--')) {
+      backgroundColor = `var(${backgroundColor})`;
```

Comment:
Since the themeObserver already provide suck function. There's no reason to manually assemble string.

Reply:
Here is some compatibility, provided that themeObserver can be obtained